scripts/run-eslint.ts

```diff
@@ -63,14 +113,37 @@ main().catch((error) => {
  * The entrypoint to this script.
  */
 async function main() {
-  const { cache, fix, quiet } = parseCommandLineArguments();
+  const { cache, fix, files: givenFiles, quiet } = parseCommandLineArguments();
+
+  const eslint = new ESLint({
+    cache,
+    errorOnUnmatchedPattern: false,
+    fix,
+    ruleFilter: ({ severity }) =>
+      !quiet || severity === ESLintMessageSeverity.Error,
+  });
+
+  const fileFilteredResults = await eslint.lintFiles(
+    givenFiles.length > 0 ? givenFiles : ['.'],
+  );
+
+  const filteredResults = quiet
+    ? ESLint.getErrorResults(fileFilteredResults)
+    : fileFilteredResults;
+
+  await printResults(eslint, filteredResults);
 
-  const eslint = new ESLint({ cache, fix });
-  const results = await runESLint(eslint, { fix, quiet });
-  const hasErrors = results.some((result) => result.errorCount > 0);
+  if (fix) {
+    await ESLint.outputFixes(fileFilteredResults);
+  }
+  const hasErrors = filteredResults.some((result) => result.errorCount > 0);
+
+  const qualityGateStatus = applyWarningThresholdsQualityGate({
+    results: filteredResults,
+  });
 
-  if (!quiet && !hasErrors) {
-    evaluateWarnings(results);
+  if (hasErrors || qualityGateStatus === QualityGateStatus.Increase) {
+    process.exitCode = 1;
```

Comment:
I pulled this out of `runESLint` and what is now `applyWarningThresholdsQualityGate` because otherwise it was hard to see.